### PR TITLE
Word Counter & Avg. Read Time Implementation 📊

### PR DESCRIPTION
**Pull request by Ismail-AD**
Hey @aritra-tech ,

Issue #116 
I've addressed the issues and added a Word Counter along with the Average Read Time functionality in this pull request. The implementation follows scientific guidelines for accurate timing.

You can review the changes and find a video attached for a quick overview.

https://github.com/aritra-tech/Notify/assets/99780212/9ec96a5b-7c5b-40d7-8fa2-e14c6321563d

Looking forward to your feedback!

### CODE REVIEW

**Comment by aritra-tech** [comment]
Please raise the PR against `develop` not `master` @Ismail-AD.

**Comment by Ismail-AD** [comment]
@aritra-tech review :)

File: app/src/main/java/com/aritra/notify/ui/screens/notes/addEditScreen/AddEditScreen.kt [hunk 1]
```diff
@@ -129,6 +130,18 @@ fun AddEditScreen(
     var photoUri by remember { mutableStateOf(emptyList<Uri?>()) }
 
     var characterCount by remember { mutableIntStateOf(title.length + description.length) }
+
+
+    val titleWordCount = remember { derivedStateOf { countWords(title) } }
+    val desWordCount = remember { derivedStateOf { countWords(description) } }
+    var totalWordCount by remember {
+        mutableIntStateOf(titleWordCount.value + desWordCount.value)
```
Comment (aritra-tech) [inline app/src/main/java/com/aritra/notify/ui/screens/notes/addEditScreen/AddEditScreen.kt:138]: The `totalWordCount` will be only the `desWrordCount` not the addition of `title` & `des`

**Review by aritra-tech — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/aritra/notify/ui/screens/notes/addEditScreen/AddEditScreen.kt [hunk 2]
```diff
@@ -148,6 +161,8 @@ fun AddEditScreen(
     }
     val formattedDateTime = SimpleDateFormat(Const.DATE_TIME_FORMAT, Locale.getDefault()).format(dateTime ?: 0)
     val formattedCharacterCount = "${(title.length) + (description.length)} characters"
+    val formattedWordCount = "${countWords(title) + countWords(description)} words"
+    val formattedReadTime = "${calculateTime((countWords(title) + countWords(description)),wordsPerMinute)} sec (reading time)"
```
Comment (aritra-tech) [inline app/src/main/java/com/aritra/notify/ui/screens/notes/addEditScreen/AddEditScreen.kt:165]: Make it only `sec read`.
e.g `1 sec read` , `6 sec read`

**Review by aritra-tech — CHANGES_REQUESTED** [review]

**Comment by aritra-tech** [comment]
And please fix the ktlint check once.

**Comment by Ismail-AD** [comment]
I will try that too if it wants to be fixed 😕 I will be back after changes @aritra-tech

**Comment by aritra-tech** [comment]
> I will try that too if it wants to be fixed 😕 I will be back after changes @aritra-tech

Sure @Ismail-AD take your time no need to hurry.

File: app/src/main/java/com/aritra/notify/ui/screens/notes/addEditScreen/AddEditScreen.kt (hunk 2 above)
Comment (Ismail-AD) [reply app/src/main/java/com/aritra/notify/ui/screens/notes/addEditScreen/AddEditScreen.kt:165]: it is already seconds read you can check out video again :) or you are requesting something else

Comment (aritra-tech) [reply app/src/main/java/com/aritra/notify/ui/screens/notes/addEditScreen/AddEditScreen.kt:165]: Currently, it is `1 sec (reading time)` I was telling you to write it as `1 sec read`

Comment (Ismail-AD) [reply app/src/main/java/com/aritra/notify/ui/screens/notes/addEditScreen/AddEditScreen.kt:165]: gotcha already bro

**Pull request closed without merging**